File: 백준/Silver/1018. 체스판 다시 칠하기/체스판 다시 칠하기.cc
#include <algorithm>
#include <iostream>
#include <queue>
#include <vector>
#define ll long long
#define fastio cin.tie(0)->sync_with_stdio(0);
#define for_in(n) for (int i = 0; i < n; ++i)
#define si(x) int(x.size())
#define all(x) (x).begin(), (x).end()
#define pb(...) push_back(__VA_ARGS__)
using namespace std;

int n, m;
int c2i(char& c) { return c == 'W' ? 0 : 1; }
int f(vector<string>& a, int x, int y) {
  int ret = 0;
  for (int i = x; i < x + 8; ++i) {
    for (int j = y; j < y + 8; ++j) {
      if (((i + j) & 1) == c2i(a[i][j])) ret++;
    }
  }
  return min(ret, 64 - ret);
}
int main() {
  fastio;
  cin >> n >> m;
  vector<string> a(n);
  int ans = 0x3f3f3f3f;
  for (auto& i : a) cin >> i;
  for (int i = 0; i < n - 7; ++i) {
    for (int j = 0; j < m - 7; ++j) {
      ans = min(ans, f(a, i, j));
    }
  }
  cout << ans << "\n";
}

File: 백준/Silver/1018. 체스판 다시 칠하기/체스판 다시 칠하기.py
def s2i(s):
  if s == "W": return 0
  return 1
def f(x, y):
  ret = 0
  for i in range(x, x + 8):
    for j in range(y, y + 8):
      if (i + j) % 2 != s2i(board[i][j]): ret += 1
  
  return min(ret, 64 - ret)

n, m = map(int, input().split())
board = [input() for i in range(n)]
ans = 0x3f3f3f3f
for i in range(n - 7):
  for j in range(m - 7):
    ans = min(ans, f(i, j))

print(ans)